### PR TITLE
chore: lift compontent logic out of the db and start exposing models

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

lift compontent logic out of the db and start exposing models

### Why is it needed?

offer more granularity for internal work

### How to test it?

make sure all the tests are passing and you can keep using the core store / webhooks & components / dynamic zones

### CODE REVIEW

File: packages/core/core/src/Strapi.ts [hunk 1]
```diff
@@ -481,16 +481,16 @@ class Strapi extends Container implements StrapiI {
...
       ...this.config.get('database'),
-      models: utils.transformContentTypesToModels(contentTypes),
+      models: [
+        utils.transformContentTypesToModels([
+          ...Object.values(this.contentTypes),
+          ...Object.values(this.components),
+        ]),
+        coreStoreModel,
+        webhookModel,
+      ],
```
Comment (Marc-Roig) [inline packages/core/core/src/Strapi.ts:493]: the ideea is to provide an interface to register models dynamically here too right?

Comment (alexandrebodin) [reply packages/core/core/src/Strapi.ts:493]: yes eventually :)

File: packages/core/database/src/metadata/index.ts [hunk 2]
```diff
@@ -36,46 +36,26 @@ export const createMetadata = (models: Model[] = []): Metadata => {
 
   // init pass
   for (const model of _.cloneDeep(models)) {
-    if ('id' in model.attributes) {
-      throw new Error('The attribute "id" is reserved and cannot be used in a model');
-    }
```
Comment (Marc-Roig) [inline packages/core/database/src/metadata/index.ts:41]: should we prevent this somewhere else? (pbbly in the transform-content-types-to-models utils) There is a reserved attribute names in there

**Review by Marc-Roig — APPROVED** [review]
🚀  looks good !

File: packages/core/database/src/metadata/index.ts (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/database/src/metadata/index.ts:41]: Sure I can readd the check for the CTs 👍

Comment (alexandrebodin) [reply packages/core/database/src/metadata/index.ts:41]: Update done :)

**Pull request merged**